### PR TITLE
uncontained.io theme

### PR DESCRIPTION
**Pull request by rdebeasi**
## What is this PR About?
This PR creates a new theme for uncontained.io. This gives us a "clean slate" to begin working from and avoids the problems and complexity that would come from "forking" an existing theme.

screenshot-2018-4-18 home uncontained io

The theme [readme](https://github.com/redhat-cop/uncontained.io/blob/28e5d8ffbeb247ced715219934125c2eedfb3903/themes/uncontained.io/README.md) has more information, but here is an overview of what's included:
- Basic functionality from [Blank theme](https://github.com/Vimux/blank)
- Simplified Bootstrap styles
    - Includes the functionality we need, taking the stylesheet from 140k to about 36k
    - Configured to follow typography and color guidelines from the branding guide
- Simple Gulp build process
    - Includes scss compilation/minification and source maps. JS/Babel could be added in the future.

The headings as specified in the brand guide are pretty huge - we can discuss and see whether we like those sizes.

### Next steps
- Homepage layout
- Revise article layout
- Create guide layout
- Add the logo once one is chosen
- Build out search (using [fuse](https://gist.github.com/eddiewebb/735feb48f50f0ddd65ae5606a1cb41ae) or something similar)
- Figure out how navigation and "sidebar" should look
- Customize pagination

## How should we test or review this PR?
- Follow the steps in the theme readme and confirm that you're able to get up and running with theme development.
- Review the code and decide whether this is a reasonable starting point for future theme development.

## Is there a relevant Trello card or Github issue open for this?
[Create a clean, no-styling uncontained theme we can use as a base with Brand](https://trello.com/c/cpvCeYzZ/114-create-a-clean-no-styling-uncontained-theme-we-can-use-as-a-base-with-brand)

## Who would you like to review this?
cc: @redhat-cop/cant-contain-this

-----

- [x] Have you followed the [contributing guidelines](https://github.com/redhat-cop/uncontained.io/blob/master/CONTRIBUTING.md)?
- [x] Have you explained what your changes do, and why they add value to the uncontained.io guides?

-----

**Please note: we may close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

File: themes/uncontained.io/package.json [hunk 1]
```diff
@@ -1,37 +1,26 @@
...
-    "start": "hugo server --source=exampleSite --themesDir=../.. --disableFastRender",
-    "watch": "webpack --watch --progress --colors"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "watch": "gulp watch",
+    "build": "gulp",
+    "start": "gulp watch"
   },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/redhat-cop/uncontained.io/tree/master/themes/uncontained.io"
```
Comment (etsauer) [inline themes/uncontained.io/package.json]: What happens with this URL? Do we need to consider support for multi-branch when we go to do pipelines?

File: themes/uncontained.io/theme.toml [hunk 2]
```diff
@@ -1,36 +1,11 @@
...
+description = "Hugo Theme for uncontained.io. Based on Blank theme by Vimux."
+homepage = "http://uncontained.io/"
+features = ["blog"]
+min_version = 0.19
 
 [author]
-  name = "Munif Tanjim"
-  homepage = "https://muniftanjim.com"
+  name = "Red Hat Communities of Practice"
+  homepage = "https://github.com/redhat-cop/"
```
Comment (etsauer) [inline themes/uncontained.io/theme.toml]: Let's set this to redhat-cop.github.io (in progress)

File: themes/uncontained.io/layouts/partials/sidebar.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+	<aside>
+		<div>
+			<div>
+				<h3>Latest Posts</h3>
+			</div>
+			<div>
+				<ul>
+					{{ range first 5 .Site.Pages }}
```
Comment (etsauer) [inline themes/uncontained.io/layouts/partials/sidebar.html]: I think we need to target specific sections with this sidebar layout. Right now I get a list of pages that includes:

Home
Content Types
Guides
Articles
Openshift LDAP Integration

File: themes/uncontained.io/package.json (hunk 1 above)
Comment (rdebeasi) [reply themes/uncontained.io/package.json]: This URL is just there for humans who are reading the file and want to learn more. It's not used by any automated process, so we could change it down the road without breaking anything. ([More info on the git attribute in package.json](redhat-cop.github.io))

File: themes/uncontained.io/layouts/partials/sidebar.html (hunk 3 above)
Comment (rdebeasi) [reply themes/uncontained.io/layouts/partials/sidebar.html]: Yeah, this feature is kind of weird - it came as part of the starter theme and isn't super helpful as-is. I'd propose removing the "sidebar" for now, and then adding more useful navigation in the future. Sound ok?

File: themes/uncontained.io/theme.toml (hunk 2 above)
Comment (rdebeasi) [reply themes/uncontained.io/theme.toml]: Sure, all set!

File: themes/uncontained.io/layouts/partials/sidebar.html (hunk 3 above)
Comment (rdebeasi) [reply themes/uncontained.io/layouts/partials/sidebar.html]: Cool! I've made that change.

**Comment by rdebeasi** [comment]
I think I've addressed the comments so far. If you have any other feedback, just let me know. Thanks for taking a look! :smiley:

**Pull request merged**